src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -79,24 +79,13 @@ public static void SetConfirmationId(this IBasicProperties properties, ulong con
         public static bool TryGetConfirmationId(this IBasicProperties properties, out ulong confirmationId)
         {
             confirmationId = 0;
-
-            if (properties.Headers.ContainsKey(confirmationIdHeader))
-            {
-                var headerBytes = properties.Headers[confirmationIdHeader] as byte[];
-                var headerString = Encoding.UTF8.GetString(headerBytes ?? new byte[0]);
-
-                return UInt64.TryParse(headerString, out confirmationId);
-            }
-
-            return false;
+            object value;
+            return properties.Headers.TryGetValue(confirmationIdHeader, out value) &&
+                ulong.TryParse(Encoding.UTF8.GetString(value as byte[] ?? new byte[0]), out confirmationId);
         }
 
-        static bool TryGet<T>(List<DeliveryConstraint> list, out T constraint) where T : DeliveryConstraint
-        {
-            constraint = list.OfType<T>().FirstOrDefault();
-
-            return constraint != null;
-        }
+        static bool TryGet<T>(this List<DeliveryConstraint> list, out T constraint) where T : DeliveryConstraint
+            => (constraint = list.OfType<T>().FirstOrDefault()) != null;
```

Comment:
Again, I'd prefer to keep it on two lines if that's OK. One line would be 174 chars long.

However, in revisiting this line I noticed that the expression operator `=>` should really be on the first line. That seems to be the more common convention.